starknet_py/utils/typed_data.py

```diff
@@ -360,6 +398,66 @@ def encode_bool(value: Union[bool, str, int]) -> int:
     raise ValueError(f"Expected boolean value, got [{value}].")
 
 
+def is_digit_string(s: str, signed=False) -> bool:
+    if signed:
+        return bool(re.fullmatch(r"-?\d+", s))
+    return bool(re.fullmatch(r"\d+", s))
+
+
+def encode_u128(value: Union[str, int]) -> int:
+    def is_in_range(n: int):
+        return 0 <= n < 2**128
+
+    if isinstance(value, int):
+        if is_in_range(value):
+            return value
+        raise ValueError(f"Value [{value}] is out of range for '{BasicType.U128}'.")
+
+    if isinstance(value, str):
+        int_value = None
+
+        if value.startswith("0x"):
+            int_value = int(value, 16)
+        elif is_digit_string(value):
+            int_value = int(value)
+        else:
+            raise ValueError(f"Value [{value}] is not a valid number.")
+
+        if int_value is not None and is_in_range(int_value):
+            return int_value
```

Comment:
Maybe something like this? The point is to avoid duplicating the same error twice and to throw `not valid number` error for `None` as well. We should apply similar modifications to `i128`.
```suggestion
    if isinstance(value, str) and value.startswith("0x"):
        int_value = int(value, 16)
    elif isinstance(value, str) and is_digit_string(value):
        int_value = int(value)
    elif isinstance(value, int):
        int_value = value
    else:
        raise ValueError(f"Value [{value}] is not a valid number.")

    if is_in_range(int_value):
        return int_value
```